### DISCUSSION TITLE
About the Implementation of RangeSearch

### DISCUSSION THREAD

**Discussion by gouyt13** [milvus-io/milvus#28921] [Q&A and General discussion] [answered]
I found in the VectorMemIndex.cpp that the index_ has a function "RangeSearch", but I cannot find the the source code of the function.

> auto res = index_.RangeSearch(*dataset, search_config, bitset);
[https://github.com/milvus-io/milvus/blob/342635ed6124916f54b64395eed9237ab3d2a21f/internal/core/src/index/VectorMemIndex.cpp](url)

**Comment by xiaofan-luan** [comment, marked as answer]
The range search is actually at

https://github.com/zilliztech/knowhere

this is the core vector search engine of milvus

**Reply by gouyt13** [reply]
Thank you. I found it!
